### PR TITLE
fixes #36, upgrade to terraform version 0.12

### LINKED ISSUE #36: Error trying to add a new remote metastore
I add the new metastore to the remote_metastores list (in this case going from 2 metastores to 3 metastores), but running plan or apply yields: 

```
Error: Error refreshing state: 1 error(s) occurred:

* module.apiary-federation.data.template_file.remote_metastores_yaml: 1 error(s) occurred:

* module.apiary-federation.data.template_file.remote_metastores_yaml[2]: index 2 out of range for list data.external.endpoint_dnsnames.*.result (max 2) in:

${lookup(data.external.endpoint_dnsnames.*.result[count.index],"dnsname")}
```

### PR DESCRIPTION
**Pull request by rpoluri**

### CODE REVIEW

File: common.tf [hunk 1]
```diff
@@ -13,15 +13,15 @@ locals {
 data "aws_caller_identity" "current" {}
 
 data "aws_vpc" "waggledance_vpc" {
-  id = "${var.vpc_id}"
+  id = var.vpc_id
 }
 
 data "aws_secretsmanager_secret" "bastion_ssh_key" {
   count = "${var.bastion_ssh_key_secret_name == "" ? 0 : 1}"
```
Comment (barnharts4) [inline common.tf]: I'm confused as to why some variable references throughout this PR are still using the `${var.name}" convention.  If that method of referencing vars still works in Terraform 0.12, why did you need to change the majority of them?

Comment (rpoluri) [reply common.tf]: new code is not backward compatible because of this line, main reason for this PR
https://github.com/ExpediaGroup/apiary-federation/blob/c7f6b7b0b628d6e8781eff6a68622f4bde4db769/endpoints.tf#L49

since it is not backward compatible anyway, updated most code to use recommended way of using variables in TF 0.12, main change in TF 0.12 is to support data types other than strings for parameters and variables, so it is not needed to enclose everything in `"${......}"`

Comment (barnharts4) [reply common.tf]: The OCD in me hates that some of the variable references are still in old, pre-0.12 format.  Unless there is a compelling reason to keep them, can you change them all?  Otherwise, it is just confusing to people who haven’t dealt with different versions of Terraform.

Comment (rpoluri) [reply common.tf]: sounds good, I will have another pass.
in general, we can use new format for most variables, might need to use old format when we are construction string like `"prefix-${var.variable}-suffix"`
